emcc.py

```diff
@@ -1132,13 +1132,19 @@ def check(input_file):
       shared.Settings.FILESYSTEM = 0
       shared.Settings.FETCH = 1
       options.js_libraries.append(shared.path_from_root('src', 'library_asmfs.js'))
+      # library_asmfs.js requires emscripten_asmfs_readdir
+      shared.Settings.EXPORTED_FUNCTIONS += ['_emscripten_asmfs_readdir']
+      shared.Settings.EXPORTED_FUNCTIONS += ['___errno_location']
 
```

Comment:
the changes in this file seem unrelated to the PR?

Reply:
These changes are necessary to support the changes in `library_asmfs.js` related to `readdir()`. The changes below this aren't specifically related to this PR but necessary for the tests to pass. I yoinked them from #8454